esi_leap/common/flocx_market.py

```diff
@@ -0,0 +1,110 @@
+from esi_leap.common import statuses
+from esi_leap.db import api as dbapi
+from esi_leap.objects import flocx_market_client
+from esi_leap.objects import offer
+import json
+from keystoneauth1 import adapter
+from keystoneauth1 import loading as ks_loading
+from oslo_config import cfg
+import datetime
+
+CONF = cfg.CONF
+
+def encode_hook(pairs):
+    new_pairs = {}
+    for key, value in pairs.iteritems():
+        if isinstance(value, unicode):
+            value = value.encode('utf-8')
+        if isinstance(key, unicode):
+            key = key.encode('utf-8')
+        new_pairs[key] = value
+    return new_pairs
+
+def get_flocx_market_client():
+    auth_plugin = ks_loading.load_auth_from_conf_options(
+        CONF, 'flocx_market')
+    sess = ks_loading.load_session_from_conf_options(CONF, 'flocx_market',
+                                                         auth=auth_plugin)
+    adpt = adapter.Adapter(
+        session=sess,
+        service_type='marketplace',
+        interface='public')
+    return flocx_market_client.FlocxMarketClient(adpt)
+
+def retrieve_from_flocx_market(context):
+    marketplace_client = get_flocx_market_client()
+    contract_list = get_contracts(context, marketplace_client)
+    return contract_list
+
+def get_contracts(context, marketplace_client):
+    contracts = []
+    contract_offer_list = get_ocr_by_status(marketplace_client)
+    if len(contract_offer_list) == 0:
+        return []
+    for contract_offer in contract_offer_list:
+        contract_data = {}
+        contract_id = contract_offer["contract_id"]
+        marketplace_offer_id = contract_offer["marketplace_offer_id"]
+        ocr_id = contract_offer["offer_contract_relationship_id"]
+        contract_data["marketplace_offer_contract_relationship_id"] = ocr_id 
+        c = get_contract_by_id(contract_id, marketplace_client)
+        contract_data["start_time"] = c["start_time"]
```

Comment:
This should be `contract_data["start_date"] = c["start_time"]`; we have different column names, apparently. Same for the `end_time` on the next line.

Reply:
My bad! Fixed.